docs/content/en/docs/how-tos/debug/_index.md

```diff
@@ -8,35 +8,112 @@ This page describes `skaffold debug`, a zero-configuration solution for
 setting up containers for debugging on a Kubernetes cluster. 
 
 {{< alert title="Note" >}}
-This functionality is in an alpha state and may change without warning.
+The `debug` functionality is in an alpha state and may change without warning.
 {{< /alert >}}
 
 ## Debugging with Skaffold
 
 `skaffold debug` acts like `skaffold dev`, but it configures containers in pods
 for debugging as required for each container's runtime technology.
-The associated debugging ports are exposed and labelled and port-forwarded to the
+The associated debugging ports are exposed and labelled so that they can be port-forwarded to the
 local machine.  Helper metadata is also added to allow IDEs to detect the debugging
 configuration parameters.
  
 ## How it works
 
-`skaffold debug` examines the built artifacts to determine the underlying runtime technology
-(currently supported: Java, NodeJS, and Python).  Any Kubernetes manifest that references these
-artifacts are transformed to enable the runtime technology's debugging functions:
+`skaffold debug` examines the built artifacts to determine the underlying runtime technology.
+Any Kubernetes manifest that references these artifacts are transformed to enable the runtime technology's
+debugging functions.
 
-  - a JDWP agent is configured for Java applications,
-  - the Chrome DevTools inspector is configured for NodeJS applications,
-  - Python applications are configured to use [`ptvsd`](https://github.com/microsoft/ptvsd/).
+`skaffold debug` uses a set of heuristics to identify the runtime technology.  
+The Kubernetes manifests are transformed on-the-fly such that the on-disk representations
+are untouched.
 
-`skaffold debug` uses a set of heuristics to identify the runtime
-technology.  The Kubernetes manifests are transformed on-the-fly
-such that the on-disk representations are untouched.
+Each Pod will have an `debug.cloud.google.com/config` annotation with a JSON object
+describing the debug configurations for the pod's containers (linebreaks for readability):
+```  
+	debug.cloud.google.com/config={
+		"<containerName>":{"runtime":"<runtimeId>",...},
+		"<containerName>":{"runtime":"<runtimeId>",...},
+		}
+```
+
+For example the following annotation indicates that the container named `web` is a Go application
+that is being debugged by a headless Delve session on port `56268`:
+```
+debug.cloud.google.com/config={"web":{"dlv":56268,"runtime":"go"}}
```

Comment:
I find this configuration a little odd? `"dlv"` describes the debugging mechanism and also the port reference? I would expect something like `{"runtime":"go", "debugger":"delve", "port":56268}`

Reply:
The `dlv` is really the protocol.  If Delve were to support the debug adapter protocol, then we could expose it instead with `"dap":xxxx`.  Some setups, like the Python `ptvsd`, may actually be able to support two protocols: a "dap" and the underlying native "pydevd" protocol, in which case it would make sense to allow `...,"dap":xxx,"pydevd":yyy,...` (untested at the moment).

Reply:
Just curious if should "dlv" could be a map to support further customization? Or is it not expected that dlv is further configured here?

```
"dlv": {
  "port": 56268
}
```

Reply:
This structure reports the configuration details to the IDE for it to make the connection to the IDE.  This stuff is alpha and liable to change, so if we need to change it we can :-)  Our IDEs are bundling specific versions of Skaffold so they should be insulated.